test/integration/infrastructure/infrastructure_test.go

```diff
@@ -451,21 +469,70 @@ func teardownNetwork(ctx context.Context, logger logr.Logger, project string, co
 
 	routerOp, err := computeService.Routers.Delete(project, *region, routerName).Context(ctx).Do()
 	if err != nil {
-		return err
-	}
-
-	logger.Info("Waiting until router is deleted...", "router", routerName)
-	if err := waitForOperation(ctx, project, computeService, routerOp); err != nil {
-		return err
+		if isNotFoundError(err) {
```

Comment:
Cosmetic suggestion:

```suggestion
		if gcpclient.IsErrorCode(err, http.StatusNotFound) {
```

You could reuse the already existing func https://github.com/gardener/gardener-extension-provider-gcp/blob/bbaf3eea152e6192d3c1bb7a735a9627e04ca5d3/pkg/gcp/client/errors.go#L21-L39 and drop the `isNotFoundError` func.